x-pack/plugins/upgrade_assistant/server/lib/es_migration_apis.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+
+import { i18n } from '@kbn/i18n';
+import {
+  CallClusterWithRequest,
+  DeprecationAPIResponse,
+  DeprecationInfo,
+  Request,
+} from 'src/core_plugins/elasticsearch';
+
+import { CURRENT_MAJOR_VERSION } from '../../common/version';
+
+import fakeDeprecations from './__fixtures__/fake_deprecations.json';
+
+export interface EnrichedDeprecationInfo extends DeprecationInfo {
+  index?: string;
+  node?: string;
+  actions?: Array<{
+    label: string;
+    url: string;
+  }>;
+}
+
+export interface UpgradeAssistantStatus {
+  cluster: EnrichedDeprecationInfo[];
+  nodes: EnrichedDeprecationInfo[];
+  indices: EnrichedDeprecationInfo[];
+
+  [checkupType: string]: EnrichedDeprecationInfo[];
+}
+
+export async function getUpgradeAssistantStatus(
+  callWithRequest: CallClusterWithRequest,
+  req: Request,
+  basePath: string,
+  useFakeData = false // TODO: remove
+): Promise<UpgradeAssistantStatus> {
+  let deprecations: DeprecationAPIResponse;
+
+  // Fake data for now. TODO: remove this before merging.
+  if (useFakeData) {
+    deprecations = _.cloneDeep(fakeDeprecations) as DeprecationAPIResponse;
+
+    // trigger pagination on cluster tab
+    for (let i = 0; i < 6; i++) {
+      deprecations.cluster_settings = deprecations.cluster_settings.concat(
+        deprecations.cluster_settings.map(cs => ({ ...cs, message: `${cs.message} ${i}` }))
+      );
+    }
+  } else {
+    deprecations = await callWithRequest(req, 'transport.request', {
+      path: '/_xpack/migration/deprecations',
+      method: 'GET',
+    });
+  }
+
+  return {
+    cluster: deprecations.cluster_settings,
+    nodes: deprecations.node_settings,
+    indices: getCombinedIndexInfos(deprecations, basePath),
+  };
+}
+
+// Combines the information from the migration assistance api and the deprecation api into a single array.
+// Enhances with information about which index the deprecation applies to and adds buttons for accessing the
+// reindex UI.
+const getCombinedIndexInfos = (deprecations: DeprecationAPIResponse, basePath: string) => {
+  const combinedIndexInfo: EnrichedDeprecationInfo[] = [];
+
+  Object.keys(deprecations.index_settings).forEach(indexName => {
+    deprecations.index_settings[indexName]
+      .map(d => ({ ...d, index: indexName } as EnrichedDeprecationInfo))
+      .map(d => {
+        // Add reindexing action if it's the old index deprecation warning.
+        if (d.message === `Index created before ${CURRENT_MAJOR_VERSION}.0`) {
```

Comment:
Is this still the message from ES for when a reindex is required?

As an aside, I am not seeing indices required for reindex being returned by the ES deprecations API. I don't see that being tracked in https://github.com/elastic/elasticsearch/issues/36024. Do we know what the status of that is?

Reply:
Yes this is still the message. I was able to get indices created in 5.6 returned from ES as old from a 6.6 snapshot. Haven't tried against a 7.0.0 snapshot actually.

Reply:
👍 I tested an index created in 6.6 against 7.0 - will try 5.6-> 6.6.